### PR TITLE
update ruby version and gemfiles

### PR DESCRIPTION
**Pull request by activemq**
Still requires some azure pipelines work

### CODE REVIEW

**Comment by GuyHarwood** [comment]
the build server dockerfile and other dependencies will need bumping in line with this, as they have a ruby dependency

File: test/admin-hpa/Rakefile [hunk 1]
```diff
@@ -42,8 +44,8 @@ array_of_features.each_with_index do |features, index|
   Cucumber::Rake::Task.new(('feature_set_' + index.to_s).to_sym, "Runs following features #{features.join(' ')}") do |t|
     File.truncate('rerun.txt', 0) if File.exist?('rerun.txt')
     @start_time = Time.now
-    t.cucumber_opts = "--tags 'not @weekend' --format pretty --retry 1 -f rerun -o rerun.txt #{features.join(' ')} DRIVER=#{(ENV['DRIVER']) ? ENV['DRIVER'] : 'headless_chrome'}"
-    t.cucumber_opts << " #{ENV['OPTS']}" unless ENV['OPTS'].nil?
+    # t.cucumber_opts = "--tags 'not @weekend' --format pretty --retry 1 -f rerun -o rerun.txt #{features.join(' ')} DRIVER=#{(ENV['DRIVER']) ? ENV['DRIVER'] : 'headless_chrome'}"
```
Comment (jon-shipley) [inline test/admin-hpa/Rakefile:47]: Why comment this out? There should be a comment, or better still delete the lines entirely if they are no longer required.

**Review by jon-shipley — COMMENTED** [review]
LGSF, but some comments in the PR

Also, there are various other files that will need the ruby version updating too:

deploy/build-agent-ruby/ruby-base.Dockerfile
deploy/docker/circleci-ruby.Dockerfile
deploy/build-agent-ruby/Dockerfile

**Comment by activemq** [comment]
Closing as new PR created #2382

**Pull request closed without merging**